source/fuzz/fact_manager/data_synonym_and_id_equation_facts.cpp

```diff
@@ -52,19 +51,24 @@ bool DataSynonymAndIdEquationFacts::OperationEquals::operator()(
 }
 
 void DataSynonymAndIdEquationFacts::AddFact(
-    const protobufs::FactDataSynonym& fact, opt::IRContext* context) {
-  // TODO(https://github.com/KhronosGroup/SPIRV-Tools/issues/3550)
-  //  Assert that ids are not irrelevant.
+    const protobufs::FactDataSynonym& fact,
+    const IrrelevantValueFacts& irrelevant_value_facts,
+    opt::IRContext* context) {
+  assert(!irrelevant_value_facts.IdIsIrrelevant(fact.data1().object()) &&
```

Comment:
Before the assertion you'll need to do:

```
(void) irrelevant_value_facts; // Keep release compilers happy.
```

Reply:
And similar in other places in the code where you pass in a sub-manager only for assertion-checking purposes.

Reply:
Done